plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,291 @@
+                    integerStatistics.getMin().ifPresent(catalogLongData::setMinimumValue);
+                    integerStatistics.getMax().ifPresent(catalogLongData::setMaximumValue);
+                });
+                statistics.getNullsCount().ifPresent(catalogLongData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogLongData::setNumberOfDistinctValues);
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.LONG.toString());
+                catalogColumnStatisticsData.setLongColumnStatisticsData(catalogLongData);
+                break;
+            case VARCHAR:
+            case CHAR:
+            case STRING:
+                StringColumnStatisticsData catalogStringData = new StringColumnStatisticsData();
+                statistics.getNullsCount().ifPresent(catalogStringData::setNumberOfNulls);
+                toMetastoreDistinctValuesCount(statistics.getDistinctValuesCount(), statistics.getNullsCount()).ifPresent(catalogStringData::setNumberOfDistinctValues);
+                catalogStringData.setMaximumLength(statistics.getMaxValueSizeInBytes().orElse(0));
+                catalogStringData.setAverageLength(getAverageColumnLength(statistics.getTotalSizeInBytes(), rowCount, statistics.getNullsCount()).orElse(0));
+                catalogColumnStatisticsData.setType(ColumnStatisticsType.STRING.toString());
+                catalogColumnStatisticsData.setStringColumnStatisticsData(catalogStringData);
+                break;
+            default:
+                throw new TrinoException(HIVE_INVALID_METADATA, "Invalid column statistics type: " + statistics);
+        }
+        return catalogColumnStatisticsData;
+    }
+
+    private static DecimalNumber bigDecimalToGlueDecimal(BigDecimal decimal)
+    {
+        Decimal hiveDecimal = new Decimal((short) decimal.scale(), ByteBuffer.wrap(decimal.unscaledValue().toByteArray()));
+        DecimalNumber catalogDecimal = new DecimalNumber();
+        catalogDecimal.setUnscaledValue(ByteBuffer.wrap(hiveDecimal.getUnscaled()));
+        catalogDecimal.setScale((int) hiveDecimal.getScale());
+        return catalogDecimal;
+    }
+
+    private static Optional<BigDecimal> glueDecimalToBigDecimal(DecimalNumber catalogDecimal)
+    {
+        Decimal decimal = new Decimal();
+        decimal.setUnscaled(catalogDecimal.getUnscaledValue());
+        decimal.setScale(catalogDecimal.getScale().shortValue());
+        return Optional.of(new BigDecimal(new BigInteger(decimal.getUnscaled()), decimal.getScale()));
+    }
+
+    private static Optional<LocalDate> dateToLocalDate(Date date)
+    {
+        long daysSinceEpoch = date.getTime() / MILLIS_PER_DAY;
+        return Optional.of(LocalDate.ofEpochDay(daysSinceEpoch));
+    }
+
+    private static Date localDateToDate(LocalDate date)
+    {
+        long millisecondsSinceEpoch = date.toEpochDay() * MILLIS_PER_DAY;
```

Comment:
the `ThriftMetastoreUtil#toMetastoreDate` doesn't look like it can handle null references for its argument since it calls `arg.toEpochDay()`. 
How do you think we should handle this particular case ?

Reply:
That method doesn't need to check since the caller only calls it if the Optional has a value.

I think you are safe too since this method isn't ever called with a possible null value. Consider this resolved.